build.sh

```diff
@@ -21,24 +23,78 @@ OL_UBI_IMAGE="${OL_UBI_IMAGE:-openliberty/open-liberty:20.0.0.10-kernel-java11-o
 
 #
 # URL at which your outer loop deploy YAML template is hosted
 #
-DEVFILE_DEPLOY_YAML_LOC="${DEVFILE_DEPLOY_YAML_LOC:-https://raw.githubusercontent.com/OpenLiberty/application-stack/master/outer-loop/0.3/app-deploy.yaml}"
+DEVFILE_DEPLOY_YAML_LOC="${DEVFILE_DEPLOY_YAML_LOC:-https://raw.githubusercontent.com/OpenLiberty/application-stack-registry/releases/outer-loop/0.4/app-deploy.yaml}"
+
+generate() {
+
+    # Base customization.
+    mkdir -p generated
+    sed -e "s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!; s!{{.STACK_IMAGE}}!$STACK_IMAGE!; s!{{.DEVFILE_DOCKERFILE_LOC}}!$DEVFILE_DOCKERFILE_LOC!; s!{{.DEVFILE_DEPLOY_YAML_LOC}}!$DEVFILE_DEPLOY_YAML_LOC!" templates/devfile.yaml > generated/devfile.yaml
+    sed -e "s!{{.BASE_OS_IMAGE}}!$BASE_OS_IMAGE!; s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!" templates/stackimage/Dockerfile > generated/stackimage-Dockerfile
+
+    # Outer loop customization of Dockerfile
+    sed -e "s!{{.STACK_IMAGE}}!$STACK_IMAGE!; s!{{.OL_UBI_IMAGE}}!$OL_UBI_IMAGE!" templates/outer-loop/Dockerfile > generated/Dockerfile
+
+}
+
+release() {
+
+    SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+    REGISTRY_HOME=$SCRIPT_DIR/../application-stack-registry
+    RELEASES_DIR=$REGISTRY_HOME/releases
+
+    # copy devfile to releases
+    DIRS="outer-loop/0.4 outer-loop/0.4.0 outer-loop/latest"
+    for d in $DIRS; do 
+        mkdir -p $RELEASES_DIR/$d 
+        cp generated/Dockerfile $RELEASES_DIR/$d 
+        # (no customization at present)
+        cp templates/outer-loop/app-deploy.yaml $RELEASES_DIR/$d
+    done
+
+    # copy devfile to releases
+    DIRS="devfile/0.4 devfile/0.4.0  devfile/latest"
+    mkdir -p $DIRS
+    for d in $DIRS; do 
+        mkdir -p $RELEASES_DIR/$d 
+        cp generated/devfile.yaml $RELEASES_DIR/$d
+        cp generated/meta.yaml $RELEASES_DIR/$d
+    done
+
+    # copy devfile to devfiles (for 'odo create')
+    cp generated/devfile.yaml $REGISTRY_HOME/devfiles/java-openliberty
+    cp templates/meta.yaml $REGISTRY_HOME/devfiles/java-openliberty
+
+    cd $REGISTRY_HOME
+    git add releases devfiles
+    git commit -m "Commit 0.4.0 release artifacts"
+}
+
 
-# Base customization.
-sed -e "s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!; s!{{.STACK_IMAGE}}!$STACK_IMAGE!; s!{{.DEVFILE_DOCKERFILE_LOC}}!$DEVFILE_DOCKERFILE_LOC!; s!{{.DEVFILE_DEPLOY_YAML_LOC}}!$DEVFILE_DEPLOY_YAML_LOC!" src/devfile.yaml > devfile.yaml
-sed -e "s!{{.BASE_OS_IMAGE}}!$BASE_OS_IMAGE!; s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!" src/stackimage/Dockerfile > stackimage/Dockerfile
+#set the action, default to generate if none passed.
+ACTION=
+if [ $# -ge 1 ]; then
+  ACTION=$1
+  shift
+fi
 
-# Outer loop customization of Dockerfile
-sed -e "s!{{.STACK_IMAGE}}!$STACK_IMAGE!; s!{{.OL_UBI_IMAGE}}!$OL_UBI_IMAGE!" src/outer-loop/Dockerfile > outer-loop/latest/Dockerfile
+case "${ACTION}" in
```

Comment:
This script will not only default an empty input to generate, but all inputs that are not release. It
Is a bit odd that "build.sh bogus" would work.

Reply:
It could always be changed later with a usage error.

Reply:
It seems that an issue should be opened to fix this if you think that is something that should be fixed later.

Reply:
I don't personally think it needs it.